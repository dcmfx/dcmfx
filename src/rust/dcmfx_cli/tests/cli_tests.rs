use assert_cmd::Command;
use insta::assert_snapshot;

#[test]
fn print() {
  let dicom_file =
    "../../../test/assets/pydicom/test_files/SC_rgb_small_odd.dcm";

  let mut cmd = Command::cargo_bin("dcmfx_cli").unwrap();
  let assert = cmd.arg("print").arg(dicom_file).assert();

  let output = String::from_utf8(assert.get_output().stdout.clone()).unwrap();
  assert_snapshot!("print", output);
}

#[test]
fn to_json() {
  let dicom_file =
    "../../../test/assets/pydicom/test_files/SC_rgb_small_odd.dcm";

  let mut cmd = Command::cargo_bin("dcmfx_cli").unwrap();
  let assert = cmd
    .arg("to-json")
    .arg(dicom_file)
    .arg("-")
    .arg("--pretty")
    .assert();

  let output = String::from_utf8(assert.get_output().stdout.clone()).unwrap();
  assert_snapshot!("to_json", output);
}

#[test]
fn modify() {
  let dicom_file = "../../../test/assets/fo-dicom/CT-MONO2-16-ankle.dcm";
  let output_file = "out.dcm";

  let assert = Command::cargo_bin("dcmfx_cli")
    .unwrap()
    .arg("print")
    .arg(dicom_file)
    .assert()
    .success();

  let output = String::from_utf8(assert.get_output().stdout.clone()).unwrap();
  assert_snapshot!("modify_before", output);

  Command::cargo_bin("dcmfx_cli")
    .unwrap()
    .arg("modify")
    .arg("--transfer-syntax")
    .arg("explicit-vr-big-endian")
    .arg("--anonymize")
    .arg(dicom_file)
    .arg(output_file)
    .assert()
    .success();

  let assert = Command::cargo_bin("dcmfx_cli")
    .unwrap()
    .arg("print")
    .arg(output_file)
    .assert()
    .success();

  let output = String::from_utf8(assert.get_output().stdout.clone()).unwrap();
  assert_snapshot!("modify_after", output);

  std::fs::remove_file(output_file).unwrap()
}
