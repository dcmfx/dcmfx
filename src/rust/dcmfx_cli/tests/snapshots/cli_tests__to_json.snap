---
source: dcmfx_cli/tests/cli_tests.rs
expression: output
---
{
  "00080008": {
    "vr": "CS",
    "Value": [
      "DERIVED",
      "SECONDARY",
      "OTHER"
    ]
  },
  "00080016": {
    "vr": "UI",
    "Value": [
      "1.2.840.10008.5.1.4.1.1.7"
    ]
  },
  "00080018": {
    "vr": "UI",
    "Value": [
      "1.2.276.0.7230010.3.1.4.8323329.1099.1521494048.423534"
    ]
  },
  "00080020": {
    "vr": "DA",
    "Value": [
      "20170101"
    ]
  },
  "00080023": {
    "vr": "DA"
  },
  "0008002A": {
    "vr": "DT"
  },
  "00080030": {
    "vr": "TM",
    "Value": [
      "120000"
    ]
  },
  "00080033": {
    "vr": "TM"
  },
  "00080050": {
    "vr": "SH"
  },
  "00080060": {
    "vr": "CS",
    "Value": [
      "OT"
    ]
  },
  "00080064": {
    "vr": "CS",
    "Value": [
      "SYN"
    ]
  },
  "00080090": {
    "vr": "PN",
    "Value": [
      {
        "Alphabetic": "Moriarty^James"
      }
    ]
  },
  "00082111": {
    "vr": "ST",
    "Value": [
      "Scaled image"
    ]
  },
  "00082112": {
  "vr": "SQ",
    "Value": [
      {
        "00080016": {
          "vr": "UI",
          "Value": [
            "1.2.840.10008.5.1.4.1.1.7"
          ]
        },
        "00080018": {
          "vr": "UI",
          "Value": [
            "1.2.826.0.1.3680043.8.498.49043964482360854182530167603505525116"
          ]
        }
      }
    ]
  },
  "00100010": {
    "vr": "PN",
    "Value": [
      {
        "Alphabetic": "Lestrade^G"
      }
    ]
  },
  "00100020": {
    "vr": "LO",
    "Value": [
      "ID1"
    ]
  },
  "00100030": {
    "vr": "DA"
  },
  "00100040": {
    "vr": "CS",
    "Value": [
      "F"
    ]
  },
  "00101010": {
    "vr": "AS",
    "Value": [
      "024Y"
    ]
  },
  "00185100": {
    "vr": "CS"
  },
  "0020000D": {
    "vr": "UI",
    "Value": [
      "1.2.826.0.1.3680043.8.498.12406831542731051035295345080039845114"
    ]
  },
  "0020000E": {
    "vr": "UI",
    "Value": [
      "1.2.826.0.1.3680043.8.498.16157229083793556332623330502397121062"
    ]
  },
  "00200010": {
    "vr": "SH",
    "Value": [
      "1"
    ]
  },
  "00200011": {
    "vr": "IS",
    "Value": [
      1
    ]
  },
  "00200013": {
    "vr": "IS",
    "Value": [
      1
    ]
  },
  "00200020": {
    "vr": "CS"
  },
  "00200060": {
    "vr": "CS"
  },
  "00204000": {
    "vr": "LT",
    "Value": [
      "Test Image with 10 rows of (255,0,0), 10 rows of (255,128,128), 10 rows of (0,255,0), 10 rows of (128,255,128), 10 rows of (0,0,255), 10 rows of (128,128,255), 10 rows of (0,0,0), 10 rows of (64,64,64), 10 rows of (192,192,192), 10 rows of (255,255,255), uncompressed"
    ]
  },
  "00280002": {
    "vr": "US",
    "Value": [
      3
    ]
  },
  "00280004": {
    "vr": "CS",
    "Value": [
      "RGB"
    ]
  },
  "00280006": {
    "vr": "US",
    "Value": [
      0
    ]
  },
  "00280008": {
    "vr": "IS",
    "Value": [
      1
    ]
  },
  "00280010": {
    "vr": "US",
    "Value": [
      3
    ]
  },
  "00280011": {
    "vr": "US",
    "Value": [
      3
    ]
  },
  "00280030": {
    "vr": "DS",
    "Value": [
      33.333333,
      33.333333
    ]
  },
  "00280100": {
    "vr": "US",
    "Value": [
      8
    ]
  },
  "00280101": {
    "vr": "US",
    "Value": [
      8
    ]
  },
  "00280102": {
    "vr": "US",
    "Value": [
      7
    ]
  },
  "00280103": {
    "vr": "US",
    "Value": [
      0
    ]
  },
  "7FE00010": {
    "vr": "OW",
    "InlineBinary": "po00po00po00P1ewP1ewP1ewnp6enp6enp6eAA=="
  }
}
